Sources/Paywalls/Components/PaywallTextComponent.swift

```diff
@@ -54,6 +53,55 @@ public extension PaywallComponent {
             self.overrides = overrides
         }
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+
+            self.type = try container.decode(ComponentType.self, forKey: .type)
+            self.text = try container.decode(LocalizationKey.self, forKey: .text)
+            self.fontName = try container.decodeIfPresent(String.self, forKey: .fontName)
+            self.fontWeight = try container.decode(FontWeight.self, forKey: .fontWeight)
+            self.color = try container.decode(ColorScheme.self, forKey: .color)
+            self.horizontalAlignment = try container.decode(HorizontalAlignment.self, forKey: .horizontalAlignment)
+            self.backgroundColor = try container.decodeIfPresent(ColorScheme.self, forKey: .backgroundColor)
+            self.size = try container.decode(Size.self, forKey: .size)
+            self.padding = try container.decode(Padding.self, forKey: .padding)
+            self.margin = try container.decode(Padding.self, forKey: .margin)
+            self.overrides = try container.decodeIfPresent(ComponentOverrides<PartialTextComponent>.self,
+                                                           forKey: .overrides)
+
+            // Decode fontSize as CGFloat or fallback to FontSize
+            // This can be removed after 2025-03-01 when we are sure no more paywalls are using this
+            if let rawFontSize = try? container.decode(CGFloat.self, forKey: .fontSize) {
+                self.fontSize = rawFontSize
+            } else if let fontSizeEnum = try? container.decode(FontSize.self, forKey: .fontSize) {
+                self.fontSize = fontSizeEnum.size
+            } else {
+                throw DecodingError.dataCorruptedError(forKey: .fontSize,
+                                                       in: container,
+                                                       debugDescription: "Invalid fontSize format")
+            }
+        }
+
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+
+            try container.encode(type, forKey: .type)
+            try container.encode(text, forKey: .text)
+            try container.encodeIfPresent(fontName, forKey: .fontName)
+            try container.encode(fontWeight, forKey: .fontWeight)
+            try container.encode(color, forKey: .color)
+            try container.encode(horizontalAlignment, forKey: .horizontalAlignment)
+            try container.encodeIfPresent(backgroundColor, forKey: .backgroundColor)
+            try container.encode(size, forKey: .size)
+            try container.encode(padding, forKey: .padding)
+            try container.encode(margin, forKey: .margin)
+            try container.encodeIfPresent(overrides, forKey: .overrides)
+
+            // Encode fontSize as CGFloat
+            try container.encode(fontSize, forKey: .fontSize)
```

Comment:
I know I'm late but (perhaps for a future PR), since this implementation of `encode(to:)` is the same the default implementation of `Encondable`, I think we can remove it. Or am I missing something?